plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -304,33 +304,8 @@ func ClusterRoles() []rbac.ClusterRole {
 				rbac.NewRule("update").Groups(legacyGroup).Resources("endpoints", "secrets", "serviceaccounts").RuleOrDie(),
 				// Needed to check API access.  These creates are non-mutating
 				rbac.NewRule("create").Groups(authenticationGroup).Resources("tokenreviews").RuleOrDie(),
-
-				rbac.NewRule("list", "watch").Groups(legacyGroup).Resources(
-					"configmaps",
-					"namespaces",
-					"nodes",
-					"persistentvolumeclaims",
-					"persistentvolumes",
-					"pods",
-					"replicationcontrollers",
-					"resourcequotas",
-					"secrets",
-					"services",
-					"serviceaccounts",
-				).RuleOrDie(),
-				rbac.NewRule("list", "watch").Groups(extensionsGroup).Resources(
-					"daemonsets",
-					"deployments",
-					"podsecuritypolicies",
-					"replicasets",
-				).RuleOrDie(),
-				rbac.NewRule("list", "watch").Groups(appsGroup).Resources("deployments").RuleOrDie(),
-				rbac.NewRule("list", "watch").Groups(batchGroup).Resources("jobs", "cronjobs").RuleOrDie(),
-				rbac.NewRule("list", "watch").Groups(appsGroup).Resources("statefulsets").RuleOrDie(),
-				rbac.NewRule("list", "watch").Groups(policyGroup).Resources("poddisruptionbudgets").RuleOrDie(),
-				rbac.NewRule("list", "watch").Groups(autoscalingGroup).Resources("horizontalpodautoscalers").RuleOrDie(),
-				rbac.NewRule("list", "watch").Groups(certificatesGroup).Resources("certificatesigningrequests").RuleOrDie(),
-				rbac.NewRule("list", "watch").Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
+				// Needed for all shared informers
+				rbac.NewRule("list", "watch").Groups("*").Resources("*").RuleOrDie(),
```

Comment:
> We add a new resource which is super powerful and the garbage collection controller shouldn't have.

Add it where - to the core? As a TPR? Via an extension apiserver? Or maybe it doesn't matter and the answer is just "yes" 😄.

To recap where we are today, in the master branch, without my PR:

1. The GC controller client is allowed to get, list, watch, patch, update, and delete */* (https://github.com/kubernetes/kubernetes/blob/7b43f922aa059abfbbd9be1840499080cb7d6f54/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go#L129)
1. The GC controller only sets up monitoring resources when it is constructed, and it doesn't refresh (e.g. to pick up new TPRs) unless I'm missing seeing it (https://github.com/kubernetes/kubernetes/blob/7b43f922aa059abfbbd9be1840499080cb7d6f54/pkg/controller/garbagecollector/garbagecollector.go#L98)

My PR grants the controller-manager the ability to list/watch */*, so all current and future shared informers can function without having to remember to modify the policy.

Maybe we need to consider looking into a way to restrict which portions of the code (controllers) are allowed to use specific shared informers? Otherwise I don't have a good idea for how to use shared informers in the GC controller without granting list/watch of */* to the controller-manager client.